x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/aggs/categorization/CategorizeTextAggregationBuilder.java

```diff
@@ -116,6 +118,13 @@ public CategorizeTextAggregationBuilder(StreamInput in) throws IOException {
         super(in);
         this.bucketCountThresholds = new TermsAggregator.BucketCountThresholds(in);
         this.fieldName = in.readString();
+        // If the coordinating node is an older version then we might still receive messages from older
+        // nodes. In this case we can send back results for this node created using the new algorithm.
+        // They won't necessarily merge well with results from other nodes, but are better than nothing.
```

Comment:
Since its experimental, I think we should throw. This is not a critical functionality, nothing systemic is using this aggregation. We should throw, especially since running in a mixed cluster environment is a temporary situation.

Reply:
The new internal agg implementation serializing back to the old coordinator will throw anyways (due to things not being read of the wire well). It would be cheaper for the users (and more obvious the cause of the issue) if we just throw.

Reply:
> The new internal agg implementation serializing back to the old coordinator will throw anyways (due to things not being read of the wire well).

It won't throw on the old coordinator with the changes of b3fe740f3dac21d207e0f42e1afd8cceab5752e6 - I changed the new nodes to serialize in the format the old nodes will expect.

But still, you're probably right that a simple "not supported in mixed cluster" exception is better as it's less likely to cause people to waste time debugging strange results.  I'll change the 4 methods to throw exceptions instead.